app/clients/github/user_client.rb

```diff
@@ -1,5 +1,14 @@
 module GitHub
   class UserClient < GitHub::Client
+    def from_matching(name:)
+      if users = search(name, per_page: 5, page: 1).items
+        names = name.downcase.split(/[ -]/).to_set
+
+        users.lazy.map { profile(_1.login) }.
+          find { names.subset?(_1.name&.downcase&.split(/[ -]/).to_set) }
+      end
```

Comment:
I haven't really stress tested this reimplementation. It worked with `GitHub::UserClient.new.from_matching name: "Kasper Timm Hansen"` and only issued one request for my profile.